### PR TITLE
this mostly works fine until now

### PR DESCRIPTION
**Pull request by MehmetTopsakal**

### CODE REVIEW

File: LICENSE [hunk 1]
```diff
@@ -672,3 +672,4 @@ may consider it more useful to permit linking proprietary applications with
 the library.  If this is what you want to do, use the GNU Lesser General
 Public License instead of this License.  But first, please read
 <https://www.gnu.org/licenses/why-not-lgpl.html>.
+
```
Comment (mrakitin) [inline LICENSE]: Where did this GPL license come from? The profile_collection repos have the BSD-3-Clause license by default.

Comment (mrakitin) [reply LICENSE]: > 100644 → 100755

I don't think the file should have the execution bit.

File: README.md [hunk 2]
```diff
@@ -1,2 +1,2 @@
 # profile_collection
-profile_collection for 28ID-D tomo endstation @ NSLS-II
+profile_collection for 28ID-2-tomo endstation @ NSLS-II
```
Comment (mrakitin) [inline README.md:2]: > 100644 → 100755

I don't think the file should have the execution bit.

File: .gitignore [hunk 3]
```diff
@@ -127,3 +127,7 @@ dmypy.json
 
 # Pyre type checker
 .pyre/
+
+
+#kate
+*.kate-swp
```
Comment (mrakitin) [inline .gitignore:133]: > 100644 → 100755

I don't think the file should have the execution bit.

File: startup/90-functions.py [hunk 4]
```diff
@@ -1,57 +1,191 @@
...
+                yield from bps.abs_set(det.cam.acquire_period, 0.1, wait=False)
+            else:
+                yield from bps.abs_set(det.cam.acquire_period, acq_time_rbv, wait=True)
+            acq_period_rbv = det.cam.acquire_period.get()
+        else:
+            if det.name == 'blackfly':
+                yield from bps.abs_set(det.cam.acquire_period, min(1,acq_period), wait=False)
+            else:
+                yield from bps.abs_set(det.cam.acquire_period, acq_period, wait=True)
+            acq_period_rbv = det.cam.acquire_period.get()
```
Comment (mrakitin) [inline startup/90-functions.py:35]: This line and a couple of occurrences above should better use `yield from bps.read()` and then you could extract the needed value from the returned dictionary.

File: startup/90-functions.py [hunk 5]
```diff
@@ -114,584 +248,854 @@ def print_det_keys(det_class):
...
-            
-        da_dark = xr.DataArray(data=img_dark.astype('float32'),
-                  coords=[np.arange(img_dark.shape[0]), np.arange(img_dark.shape[1])],
-                  dims=['pixel_y', 'pixel_x'],attrs=None
-                 )
-        ds['dark'] = da_dark
-        dark_taken = 'true'
-        
+        RE(configure_area_det(det,acq_time=det_acq_time,exposure=det_exposure_dark))
+        uid_dark = RE(count([det]),md={'task':'dark','acq_time':det_acq_time,'exposure':det_exposure_dark})[0]
```
Comment (mrakitin) [inline startup/90-functions.py]: We do not recommend calling RE from inside functions. Better to make this function a plan with `yield from configure_area_det(...)`, etc. and pass it to `RE`.

File: startup/90-functions.py [hunk 6]
```diff
@@ -114,584 +248,854 @@ def print_det_keys(det_class):
...
+    
+    
+    
+    
+    
+    
+    
+    
+    
+
```
Comment (mrakitin) [inline startup/90-functions.py]: Those many blank lines are against the PEP8 standard. Would be great to run the code through `flake8` (for style check) and `black` (for automated formatting).

**Review by mrakitin — CHANGES_REQUESTED** [review]
I have some style-related comments, and concerns about the license file and the permissions for some files (such as `755` where it should be `644`). Once fixed, it's good to go in.

File: LICENSE (hunk 1 above)
Comment (MehmetTopsakal) [reply LICENSE]: It was from initial repo creation. Removed.

File: README.md (hunk 2 above)
Comment (MehmetTopsakal) [reply README.md:2]: done

File: .gitignore (hunk 3 above)
Comment (MehmetTopsakal) [reply .gitignore:133]: done

File: startup/90-functions.py (hunk 4 above)
Comment (MehmetTopsakal) [reply startup/90-functions.py:35]: I am not sure how to do it. Can you do that for me. Thanks

File: startup/90-functions.py (hunk 5 above)
Comment (MehmetTopsakal) [reply startup/90-functions.py]: @mrakitin I will work on functions later. For now, removed those stuff

File: startup/90-functions.py (hunk 6 above)
Comment (MehmetTopsakal) [reply startup/90-functions.py]: I don't know what is flake8. Will try to figure out later.

Comment (mrakitin) [reply startup/90-functions.py]: https://flake8.pycqa.org/en/latest/. We deploy as a part of our collection environments.

File: startup/90-functions.py (hunk 4 above)
Comment (mrakitin) [reply startup/90-functions.py:35]: Let's address it via a separate PR. I would like to merge it to this repo's `main` branch shortly.

**Review by mrakitin — APPROVED** [review]

**Pull request merged**